directory/user/domain/objects.py

```diff
@@ -53,3 +57,61 @@ def do(self, email, old_password, new_password):
             raise SpecifiedUserPasswordIsIncorrectError
 
         self.user.set_new_password(email=email, password=new_password)
+
+
+class RequestUserPasswordRecovery:
+    """
+    Request to recovery user password implementation.
+    """
+
+    def __init__(self, user, password_recovery_state):
+        """
+        Constructor.
+        """
+        self.user = user
+        self.password_recovery_state = password_recovery_state
+
+    def do(self, email):
+        """
+        Request to recover user password by email.
+        """
+        if not self.user.does_exist(email=email):
+            raise UserWithSpecifiedEmailAddressDoesNotExistError
+
+        identifier = uuid.uuid4().hex
+
+        self.password_recovery_state.create(email=email, identifier=identifier)
+
+        return identifier
+
+
+class RecoveryUserPassword:
```

Comment:
`Recovery` > `Recover`. Because we write verbs, not nouns.

Reply:
@dmytrostriletskyi, done.